src/panels/lovelace/cards/hui-gauge-card.ts

```diff
@@ -1,15 +1,127 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { HomeAssistant } from "../../../types.js";
+import isValidEntityId from "../../../common/entity/valid_entity_id.js";
+import { fireEvent } from "../../../common/dom/fire_event.js";
 
 import "../../../components/ha-card.js";
 
-import EventsMixin from "../../../mixins/events-mixin.js";
+interface Config extends LovelaceConfig {
+  entity: string;
+  title?: string;
+  unit_of_measurement?: string;
+  min?: number;
+  max?: number;
+  severity?: object;
+}
+
+class HuiGaugeCard extends LitElement implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 1;
+  }
+
+  public setConfig(config) {
+    if (!config || !config.entity) {
```

Comment:
Should we limit it to sensor domain ?

Reply:
Maybe but there could be an entity that returns a numeric state thats not a sensor

Reply:
okay. As long as out configuration auto complete for this sensor will limit to sensor entities.

Reply:
Sorry doesnt this mean you want me to limit to sensor?